packages/http-specs/specs/payload/pageable/main.tsp

```diff
@@ -52,4 +52,248 @@ namespace ServerDrivenPagination {
 
     @nextLink next?: url;
   };
+
+  @route("/continuationtoken")
+  namespace ContinuationToken {
+    @scenario
+    @scenarioDoc("""
+      Test case for using continuation token as pagination. Continuation token is passed in the request body and response body.
+      
+      Two requests need to be tested.
+      1. Initial request:
+      Expected route: /payload/pageable/server-driven-pagination/continuationtoken/request-body-response-body
+      Expected response body:
+      ```json
+      { "pets": [
+          { "id": "1", "name": "dog" },
+          { "id": "2", "name": "cat" }
+        ],
+        "nextToken": "12345"
+      }
+      ```
+      2. Next page request:
+      Expected route: /payload/pageable/server-driven-pagination/continuationtoken/request-body-response-body
+      Expected request body:
+      ```json
+      { "token": "12345" }
+      ```
+      Expected response body:
+      ```json
+      { "pets": [
+          { "id": "3", "name": "bird" },
+          { "id": "4", "name": "fish" }
+        ]
+      }
+      ```
+      """)
+    @route("/request-body-response-body")
+    @list
+    op reqBodyResBody(@continuationToken token?: string): {
+      @pageItems
+      pets: Pet[];
+
+      @continuationToken nextToken?: string;
+    };
+
+    @scenario
+    @scenarioDoc("""
+      Test case for using continuation token as pagination. Continuation token is passed in the request query and response body.
+      
+      Two requests need to be tested.
+      1. Initial request:
+      Expected route: /payload/pageable/server-driven-pagination/continuationtoken/request-query-response-body
+      Expected response body:
+      ```json
+      { "pets": [
+          { "id": "1", "name": "dog" },
+          { "id": "2", "name": "cat" }
+        ],
+        "nextToken": "12345"
+      }
+      ```
+      
+      2. Next page request:
+      Expected route: /payload/pageable/server-driven-pagination/continuationtoken/request-query-response-body?token=12345
+      
+      Expected response body:
+      ```json
+      { "pets": [
+          { "id": "3", "name": "bird" },
+          { "id": "4", "name": "fish" }
+        ]
+      }
+      ```
+      """)
+    @route("/request-query-response-body")
+    @list
+    op reqQueryResBody(@continuationToken @query token?: string): {
```

Comment:
nit, prefer not to have incomplete word in name. `request` and `response`?

Reply:
Just want to keep the name as short as possible. But if other devs also prefer complete name, I am OK to update it.